sand/extensions/search/wikidata_search.py

```diff
@@ -0,0 +1,105 @@
+import requests
+from flask import request, jsonify
+from typing import Dict
+import nh3
+from sand.controllers.search import ISearch
+from sand.models.entity import Entity
+from sand.models.ontology import OntClass, OntProperty
+
+
+class WikidataSearch(ISearch):
+
+    def __init__(self):
+        self.wikidata_url = "https://www.wikidata.org/w/api.php"
+        self.local_class_idsearch_uri = "http://0.0.0.0:5525/api/classes/"
+        self.PARAMS = {
+            "action": "query",
+            "format": "json",
+            "list": "search",
+            "srsearch": "",
+            "utf8": "",
+            "srnamespace": 0,
+            "srlimit": 10,
+            "srprop": "snippet|titlesnippet"
+        }
+        self.search_item_template = {
+            "label": "",
+            "id": "",
+            "description": "",
+            "uri": "",
+        }
+
+    def get_class_search_params(self, search_text: str) -> Dict:
+        """Updates class search parameters for wikidata API"""
+        class_params = self.PARAMS.copy()
+        class_params["srnamespace"] = 0
+        class_params['srsearch'] = f"haswbstatement:P279 {search_text}"
+        return class_params
+
+    def get_local_class_properties(self, id: str) -> Dict:
+        """Calls local class search API to fetch all class metadata using class ID"""
+        api_data = requests.get(self.local_class_idsearch_uri + str(id))
```

Comment:
instead of sending requests.get to our own APIs, use `OntPropertyAR()` (this function is cached). I did notice that in `app.py`, our APIs use `ChainedMapping(OntPropertyAR(), import_attr(SETTINGS["ont_props"]["default"]))`. Perhaps we should update `OntPropertyAR` to include the default. What do you think?